pkg/cmd/cli/cmd/rsync/copyrsync.go

```diff
@@ -47,17 +48,25 @@ func newRsyncStrategy(f *clientcmd.Factory, c *cobra.Command, o *RsyncOptions) (
 	if err != nil {
 		return nil, err
 	}
-
 	// The blocking-io flag is used to resolve a sync issue when
 	// copying from the pod to the local machine
 	flags := []string{"-a", "--blocking-io", "--omit-dir-times", "--numeric-ids"}
 	flags = append(flags, rsyncFlagsFromOptions(o)...)
 
+	podNames := []string{}
+	if !o.Source.Local() {
+		podNames = append(podNames, o.Source.PodName)
+	}
+	if !o.Destination.Local() {
+		podNames = append(podNames, o.Destination.PodName)
+	}
```

Comment:
In this case, it will be either or since we don't support copying container to container... so you could have a single podName string